### PR TITLE
aeron-cpp: init at 1.42.1

### PR DESCRIPTION
**Pull request by vaci**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -18262,6 +18262,8 @@ with pkgs;
 
   aeron = callPackage ../servers/aeron { };
 
+  aeron-cpp = callPackage ../servers/aeron-cpp { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:18266]: New top-level packages should go to `pkgs/by-name` https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md
```suggestion
```

File: pkgs/servers/aeron-cpp/default.nix [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  installPhase = ''
+     mkdir --parents "$out"
+     cp --archive --verbose --target-directory="$out" install/*
+  '';
+
+  meta = {
+    description = "Aeron Messaging C++ Library";
+    homepage = "https://github.com/real-logic/aeron";
+  };
```
Comment (Lurkki14) [inline pkgs/servers/aeron-cpp/default.nix:117]: Missing `license` and `maintainer`

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (vaci) [reply pkgs/top-level/all-packages.nix]: Moved to pkgs/by-name as suggested

**Comment by vaci** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/222508

**Comment by Lurkki14** [comment]
You need to squash your commits

File: pkgs/by-name/ae/aeron-cpp/package.nix [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir --parents "$out"
+    cp --archive --verbose --target-directory="$out" install/*
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (pbsds) [inline pkgs/by-name/ae/aeron-cpp/package.nix:115]: please set `meta.sourceProvenance`

File: pkgs/by-name/ae/aeron-cpp/package.nix [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cmake
+    makeWrapper
+    patchelf
+  ];
+
+  configurePhase = ''
```
Comment (pbsds) [inline pkgs/by-name/ae/aeron-cpp/package.nix:70]: Please add `runHook preConfigure` and `runHook postConfigure`

File: pkgs/by-name/ae/aeron-cpp/package.nix [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        -DCMAKE_BUILD_TYPE=Release \
+        -DAERON_TESTS=OFF \
+        -DAERON_SYSTEM_TESTS=OFF \
+        -DAERON_BUILD_SAMPLES=OFF \
+        -DCMAKE_INSTALL_PREFIX:PATH=../../install \
+        ../..
+    )
+  '';
+
+  buildPhase = ''
```
Comment (pbsds) [inline pkgs/by-name/ae/aeron-cpp/package.nix:85]: Please add `runHook preBuild` and `runHook postBuild`

File: pkgs/by-name/ae/aeron-cpp/package.nix [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+in
+
+stdenv.mkDerivation {
+  pname = "aeron-cpp";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "real-logic";
+    repo = "aeron";
+    rev = "a6484d2796d3bc43241e4cfe48a8306f0094e5fe";
```
Comment (pbsds) [inline pkgs/by-name/ae/aeron-cpp/package.nix:44]: this repo seem to feature tags, could you use one?

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 268707` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>aeron-cpp</li>
  </ul>
</details>

File: pkgs/by-name/ae/aeron-cpp/package.nix [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  libbsd,
+  libuuid,
+  makeWrapper,
+  patchelf,
+  stdenv,
+  zlib
+}:
+
+let
+  version = "1.42.1";
```
Comment (pbsds) [inline pkgs/by-name/ae/aeron-cpp/package.nix:17]: Perhaps even 

```suggestion
  inherit (aeron) version;
```

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 268707` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>aeron-cpp</li>
  </ul>
</details>

LGTM, thanks

**Pull request merged**